OpenRA.Mods.Common/Traits/RevealsShroud.cs

```diff
@@ -18,9 +18,22 @@ public class RevealsShroudInfo : AffectsShroudInfo
 
 	public class RevealsShroud : AffectsShroud
 	{
+		readonly Actor self;
+
 		public RevealsShroud(Actor self, RevealsShroudInfo info)
-			: base(self, info) { }
-		protected override void AddCellsToPlayerShroud(Actor self, Player p, PPos[] uv) { p.Shroud.AddProjectedVisibility(self, uv); }
-		protected override void RemoveCellsFromPlayerShroud(Actor self, Player p) { p.Shroud.RemoveVisibility(self); }
+			: base(self, info) { this.self = self; }
+
+		protected override void AddCellsToPlayerShroud(object o, Player p, PPos[] uv)
+		{
+			o = this;
+			if (self.Owner.IsAlliedWith(p))
```

Comment:
It would be great if you could change this and `CreatesShroud` to use a stance check here with a TraitInfo property for who gets the reveal/hide.  This would be useful for scripted maps like allies-05, which currently teleports a camera actor over an enemy truck every tick.